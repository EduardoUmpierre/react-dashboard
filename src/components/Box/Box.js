import React from 'react'
import './Box.scss'

const Box = ({ title, children }) => (
    <div className='box'>
        <h3>{title}</h3>
        {children}
    </div>
)

export default Box
