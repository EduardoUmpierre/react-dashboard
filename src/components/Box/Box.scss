.box {
    background-color: #fff;
    border-radius: 3px;
    box-shadow: 0 1px 10px 0 rgba(0, 0, 0, 0.05);
    margin-bottom: 20px;

    > h3 {
        border-bottom: 1px solid #eee;
        color: #5f646a;
        font-size: 1.125rem;
        font-weight: 400;
        line-height: 1;
        margin-bottom: 0;
        padding: 1.5rem 0.938rem;

        > .title-icon {
            margin-right: 0.938rem;
        }
    }
}
