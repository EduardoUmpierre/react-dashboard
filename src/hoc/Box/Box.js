import React from 'react'

const Box = ({ title, children }) => (
    <div>
        <h3>{title}</h3>
        {children}
    </div>
)

export default Box
