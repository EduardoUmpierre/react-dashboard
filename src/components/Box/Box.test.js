import React from 'react'
import { shallow } from 'enzyme'
import Box from './Box'

describe('<Box />', () => {
    it('renders correctly', () => {
        const component = shallow(
            <Box title='Box title' icon='comments'>
                Test
            </Box>
        )
        expect(component).toMatchSnapshot()
    })

    it('has icon', () => {
        const component = shallow(
            <Box title='Box title' icon='comments'>
                Test
            </Box>
        )
        expect(component.exists('.title-icon')).toBe(true)
    })

    it('do not have icon', () => {
        const component = shallow(<Box title='Box title'>Test</Box>)
        expect(component.exists('.title-icon')).toBe(false)
    })
})
