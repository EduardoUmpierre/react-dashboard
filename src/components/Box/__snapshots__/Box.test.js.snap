// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box
    icon="comments"
    title="Box title"
  >
    Test
  </Box>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          <span
            className="title-icon"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={true}
              flip={null}
              icon="comments"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </span>
          Box title
        </h3>,
        "Test",
      ],
      "className": "box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="title-icon"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon="comments"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>,
            "Box title",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon="comments"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "title-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": true,
                "flip": null,
                "icon": "comments",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "Box title",
        ],
        "type": "h3",
      },
      "Test",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            <span
              className="title-icon"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon="comments"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>
            Box title
          </h3>,
          "Test",
        ],
        "className": "box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="title-icon"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={true}
                  flip={null}
                  icon="comments"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </span>,
              "Box title",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={true}
                  flip={null}
                  icon="comments"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "title-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": true,
                  "flip": null,
                  "icon": "comments",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "Box title",
          ],
          "type": "h3",
        },
        "Test",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
